cmd/spire-server/cli/federation/create.go

```diff
@@ -0,0 +1,138 @@
+package federation
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+
+	"github.com/mitchellh/cli"
+	trustdomainv1 "github.com/spiffe/spire-api-sdk/proto/spire/api/server/trustdomain/v1"
+	"github.com/spiffe/spire-api-sdk/proto/spire/api/types"
+	"github.com/spiffe/spire/cmd/spire-server/util"
+	common_cli "github.com/spiffe/spire/pkg/common/cli"
+	"google.golang.org/grpc/codes"
+)
+
+const (
+	profileHTTPSWeb    = "https_web"
+	profileHTTPSSPIFFE = "https_spiffe"
+)
+
+// NewCreateCommand creates a new "create" subcommand for "federation" command.
+func NewCreateCommand() cli.Command {
+	return newCreateCommand(common_cli.DefaultEnv)
+}
+
+func newCreateCommand(env *common_cli.Env) cli.Command {
+	return util.AdaptCommand(env, new(createCommand))
+}
+
+// FederationRelationships type is used for parsing federation relationships from file
+type federationRelationships struct {
+	FederationRelationships []*federationRelationshipJSON `json:"federation_relationships"`
+}
+
+type createCommand struct {
+	path                  string
+	trustDomain           string
+	bundleEndpointURL     string
+	bundleEndpointProfile string
+	endpointSPIFFEID      string
+	bundleFormat          string
+	bundlePath            string
```

Comment:
How would you feel about replacing these variables with an instance of the federationRelationshipJSON struct and just setting the fields directly with the flags? That removes the need for some of the conversion code below.